homeassistant/components/climate/__init__.py

```diff
@@ -717,6 +732,35 @@ def async_turn_aux_heat_off(self):
         """
         return self.hass.async_add_job(self.turn_aux_heat_off)
 
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        # For now, we convert from the legacy technique to the new
+        # supported_features. This follows exactly the way of determining
+        # whether an attribute is included in state_attributes()
+
+        support = (SUPPORT_CURRENT_TEMPERATURE | SUPPORT_TARGET_TEMPERATURE)
+        if self.target_temperature_high is not None:
+            support |= SUPPORT_TARGET_TEMPERATURE_HIGH
+            support |= SUPPORT_TARGET_TEMPERATURE_LOW
+        if self.target_humidity:
+            support |= SUPPORT_CURRENT_HUMIDITY
+            support |= SUPPORT_TARGET_HUMIDITY
+        if self.current_fan_mode is not None:
+            support |= SUPPORT_FAN_MODE
+        if self.current_operation is not None:
```

Comment:
`current_operation` reflects the state of the entity so I would assume it is a mandatory feature.

Reply:
Currently, at least the mqtt platform can come without operation mode, if you don't set the according topics. We could make the topics mandatory. However, I think there might be devices (also outside of mqtt), where you can't chose any mode, but just set a temperature?

The current front end also handles the operation mode as optional, and only displays the selection box if `operation_list` is set.

Reply:
If you can set a temperature you should be able to know what state the device is in, right? It is not mandatory to be able to change the state, ie use the `set_operation_mode` service.

Reply:
Ah, now I get it, the problem is `current_operation` vs. `set_operation_mode`. Will change that!

Reply:
So, I wonder now how to detect whether the service if offered or not. Would it be okay to still check for `self.current_operation` here? The other solution would be to call `self.set_operation_mode(self.current_operation)` and check for a `NotImplemented` exception, but that feels very unclean.

Obviously, platforms *should* override this `supported_features` method and set the supported features without such a hack. This method is basically just a legacy workaround such that all platforms keep working.